django/db/models/aggregates.py

```diff
@@ -31,11 +58,39 @@ def default_alias(self):
         expressions = self.get_source_expressions()
         if len(expressions) == 1 and hasattr(expressions[0], 'name'):
             return '%s__%s' % (expressions[0].name, self.name.lower())
-        raise TypeError("Complex expressions require an alias")
+        raise TypeError('Complex expressions require an alias')
 
     def get_group_by_cols(self):
         return []
 
+    def as_sql(self, compiler, connection, **extra_context):
+        if self.filter:
+            if connection.features.supports_aggregate_filter_clause:
+                filter_sql, filter_params = self.filter.as_sql(compiler, connection)
+                template = self.filter_template % extra_context.get('template', self.template)
+                sql, params = super().as_sql(compiler, connection, template=template, filter=filter_sql)
+                return sql, params + filter_params
+            else:
+                copy = self.copy()
+                copy.filter = None
+                condition = When(Q())
+                source_expressions = copy.get_source_expressions()
+                condition.set_source_expressions([self.filter, source_expressions[0]])
+                copy.set_source_expressions([Case(condition)] + source_expressions[1:])
+                return super(Aggregate, copy).as_sql(compiler, connection, **extra_context)
+
+        return super().as_sql(compiler, connection, **extra_context)
+
+    def __repr__(self):
+        if self.filter is None:
+            return super().__repr__()
+
+        return '{}({}, filter={})'.format(
```

Comment:
Shouldn't be part of this PR, but it looks like redundancy in `__repr__()` methods would be a good candidate for a refactor.